client/src/components/Mindblower.vue

```diff
@@ -1,17 +1,48 @@
 <template>
-  <div id="mindblower">
-    <div id="mindblower__effect-container"></div>
+  <div id="mindblower" ref="mindblower">
+    <div id="mindblower__effect-container" v-html="mindBlowerHTML" />
     <div class="mindblower__label" aria-busy="true" role="alert"></div>
   </div>
 </template>
 
-<script lang="ts">
+<script>
 export default {
   name: 'mindblower',
+  data: () => ({
+    mindBlowerHTML: '',
+  }),
+  mounted() {
+    this.mindBlowerHTML = (function() {
+      let circles = ''
+      const total = 80 // number of overlapping circles
+      const size = 1200 // diameter of circles (px)
+
+      const top = index =>
+        -(size / 2) + (size / 2) * Math.cos((2 * Math.PI * (index - 1)) / total)
+
+      const left = index =>
+        -(size / 2) + (size / 2) * Math.sin((2 * Math.PI * (index - 1)) / total)
+
+      for (
+        let i = 1, end = total, asc = 1 <= end;
+        asc ? i <= end : i >= end;
+        asc ? i++ : i--
```

Comment:
wouldn't it make more sense if you're simply do

```
Array(i).fill().map((_, i) => {
const pos = asc ? total - i : i;

// your string stuff
})
```

Reply:
possibly. some of the code was copypasta (hence the unsafe html insert) and refactoring it could currently cause more harm than good.